src/annotator/test/annotation-counts-test.ts

```diff
@@ -20,7 +24,9 @@ describe('annotationCounts', () => {
     countEl2.setAttribute('data-hypothesis-annotation-count', '');
     document.body.appendChild(countEl2);
 
-    fakeCrossFrame.on = sandbox.stub().returns(fakeCrossFrame);
+    fakeCrossFrame = {
+      on: sandbox.stub().returns(fakeCrossFrame),
+    } as PortRPCMock;
```

Comment:
The very moment we start using TS on tests, we need some way to mock complex objects without having to inline all of its methods and properties.

For example, `fakeCrossFrame` is used as a replacement of `PortRPC`, but we only bother mocking its `on` method.

With TS, the object needs to be typed as `PortRPC` so that we can pass it to functions expecting that type, but we also need to be able to expose the fact that `on` is also a `SinonStub`.

For simplicity I have just used an inline casting here, but we will probably need a more sophisticated mechanism sooner or later.

In some projects I have used [ts-mockery](https://github.com/ike18t/ts-mockery). It works pretty well, and allows for things like this:

```ts
const on = sinon.stub();
const fakeCrossFrame = Mock.of<PortRPC>({ on });

// After this line, TS knows fakeCrossFrame has type PortRPC, but we can also interact with `on` as a SinonStub.
```

The main problem is that it is pretty much unmaintained, with no activity in 3 years.

Reply:
I have realized the maintainer of `ts-mockery` is still active on issues ([see this](https://github.com/ike18t/ts-mockery/issues/56#issuecomment-1404043204)), and the last commit is from 1 year ago, and not 3 as I thought.